src/frontends/tensorflow/src/op/extract_image_patches.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op_table.hpp"
+#include "openvino/op/util/attr_types.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using namespace ov::opset8;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_extract_image_patches_op(const NodeContext& node) {
+    TENSORFLOW_OP_VALIDATION(node, node.get_input_size() >= 0, "ExtractImagePatches must have at least one input.");
+    auto images = node.get_input(0);
+
+    // retrieve attributes for Conv2DBackpropInput
+    auto tf_ksizes = node.get_attribute<std::vector<int64_t>>("ksizes");
+    auto tf_strides = node.get_attribute<std::vector<int64_t>>("strides");
+    auto tf_rates = node.get_attribute<std::vector<int64_t>>("rates");
+    auto tf_padding_type = node.get_attribute<std::string>("padding");
+    ov::op::PadType auto_pad = convert_tf_padding(node, tf_padding_type);
+    TENSORFLOW_OP_VALIDATION(node,
+                             auto_pad == ov::op::PadType::SAME_UPPER || auto_pad == ov::op::PadType::VALID,
+                             "Only SAME_UPPER and VALID padding modes are supported for ExtractImagePatches.");
+
+    // prepare attributes for OpenVINO ConvolutionBackpropData
+    Shape sizes(2), rates(2);
+    Strides strides(2);
+
+    // layout for this operation is always NHWC
+    bool is_nhwc = true;
+    convert_nhwc_to_hw(true, tf_ksizes, sizes);
+    convert_nhwc_to_hw(true, tf_strides, strides);
+    convert_nhwc_to_hw(true, tf_rates, rates);
+
+    // prepare input to ExtractImagePatches
+    convert_nhwc_to_nchw(is_nhwc, images);
+
+    auto extract_image_patches = make_shared<ExtractImagePatches>(images, sizes, strides, rates, auto_pad);
+
+    // prepare output to return the original layout NHWC
+    auto extract_image_patches_output = extract_image_patches->output(0);
+    convert_nchw_to_nhwc(is_nhwc, extract_image_patches_output);
+
+    // move the original name to new ConvolutionBackpropData if original layout is NCHW
+    // move the original name to Transpose if original layout is NHWC
```

Comment:
Since `convert_nchw_to_nhwc` alters `extract_image_patches_output` variable and it is either EIP or Transpose depending on what is provided as `is_nhwc` this both comments are unnecessary.

Reply:
Done